File: query-service-web/src/main/resources/public/app/core/models/cache-info.ts
export interface CacheInfo {
  enabled: boolean;
  connected: boolean;
  type: string;
  host: string;
  port: number;
  database: number;
  keyPrefix: string;
  defaultTtlSeconds: number;
  compressionEnabled: boolean;
  failOpen: boolean;
  errorMessage?: string;
}

export interface CacheStats {
  hits: number;
  misses: number;
  errors: number;
  evictions: number;
  keyCount: number;
  memoryUsageBytes: number;
}

export interface RouteCacheStats {
  routeId: string;
  cacheEnabled: boolean;
  cacheTtlSeconds: number | null;
  routeKeyCount: number;
  globalStats: CacheStats;
}

File: query-service-cache/src/main/java/com/inovexcorp/queryservice/cache/CacheInfo.java
package com.inovexcorp.queryservice.cache;

import lombok.Builder;
import lombok.Value;

/**
 * Information about cache connection and configuration.
 */
@Value
@Builder
public class CacheInfo {
    boolean enabled;
    boolean connected;
    String type; // "redis", "noop", etc.
    String host;
    int port;
    int database;
    String keyPrefix;
    int defaultTtlSeconds;
    boolean compressionEnabled;
    boolean failOpen;
    String errorMessage;

    // Request coalescing configuration
    boolean coalescingEnabled;
    long coalescingTimeoutMs;
}
